pkg/fs/dir_nix.go

```diff
@@ -0,0 +1,93 @@
+// +build darwin dragonfly freebsd linux netbsd openbsd
+
+/*
+ * Minio Cloud Storage, (C) 2016 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fs
+
+import (
+	"os"
+	"sort"
+	"strings"
+	"syscall"
+	"unsafe"
+)
+
+// actual length of the byte array from the c - world.
+func clen(n []byte) int {
+	for i := 0; i < len(n); i++ {
+		if n[i] == 0 {
+			return i
+		}
+	}
+	return len(n)
+}
+
+func parseDirents(buf []byte) []fsDirent {
+	bufidx := 0
+	dirents := []fsDirent{}
+	for bufidx < len(buf) {
+		dirent := (*syscall.Dirent)(unsafe.Pointer(&buf[bufidx]))
+		bufidx += int(dirent.Reclen)
```

Comment:
I think we need to add
```
		if dirent.Ino == 0 { // File absent in directory.
			continue
		}
```
you can refer https://golang.org/src/syscall/syscall_linux.go?s=#L773

Also I see we need to handle specifically for various platforms